src/FieldWarning/Assets/Camera/SlidingCameraBehaviour.cs

```diff
@@ -269,4 +332,74 @@ private void ClampCameraXZPosition()
                 Terrain.activeTerrain.GetPosition().z - _maxCameraHorizontalDistanceFromTerrain,
                 Terrain.activeTerrain.GetPosition().z + Terrain.activeTerrain.terrainData.size.z + _maxCameraHorizontalDistanceFromTerrain);
     }
+
+    private void SetPanningCursor(ScreenCorner corner)
+    {
+        Cursor.visible = false;
+        DisableAllPanningArrows();
+        switch (corner) {
+        case ScreenCorner.TopLeft:
+            _cornerArrowTopLeft.transform.position = Input.mousePosition;
+            _cornerArrowTopLeft.enabled = true;
+            break;
+        case ScreenCorner.TopRight:
+            _cornerArrowTopRight.transform.position = Input.mousePosition;
+            _cornerArrowTopRight.enabled = true;
+            break;
+        case ScreenCorner.BottomLeft:
+            _cornerArrowBottomLeft.transform.position = Input.mousePosition;
+            _cornerArrowBottomLeft.enabled = true;
+            break;
+        case ScreenCorner.BottomRight:
+            _cornerArrowBottomRight.transform.position = Input.mousePosition;
+            _cornerArrowBottomRight.enabled = true;
+            break;
+        case ScreenCorner.Top:
+            _sideArrowTop.transform.position = Input.mousePosition;
+            _sideArrowTop.enabled = true;
+            break;
+        case ScreenCorner.Bottom:
+            _sideArrowBottom.transform.position = Input.mousePosition;
+            _sideArrowBottom.enabled = true;
+            break;
+        case ScreenCorner.Left:
+            _sideArrowLeft.transform.position = Input.mousePosition;
+            _sideArrowLeft.enabled = true;
+            break;
+        case ScreenCorner.Right:
+            _sideArrowRight.transform.position = Input.mousePosition;
+            _sideArrowRight.enabled = true;
+            break;
+        case ScreenCorner.None:
+            Cursor.visible = true;
```

Comment:
Creating canvas elements and moving them to the mouse seems needlessly complicated, is there a specific reason you didn't just change the cursor texture directly with Cursor.SetCursor()? There is an example of that in InputManager.cs if you weren't aware that it's an option.

Reply:
Yes there is a specific reason. I did change the cursor at first but it would break other game mechanics like when you start a fast move command, the surcor change, but if you pan before doing the fast move command then you lose your cursor.